src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncContext.Unix.cs

```diff
@@ -610,55 +936,25 @@ public SocketError Accept(byte[] socketAddress, ref int socketAddressLen, int ti
             Debug.Assert(timeout == -1 || timeout > 0, $"Unexpected timeout: {timeout}");
 
             SocketError errorCode;
-            if (SocketPal.TryCompleteAccept(_socket, socketAddress, ref socketAddressLen, out acceptedFd, out errorCode))
+            int observedSequenceNumber;
+            if (_receiveQueue.IsReady(this, out observedSequenceNumber) &&
+                SocketPal.TryCompleteAccept(_socket, socketAddress, ref socketAddressLen, out acceptedFd, out errorCode))
```

Comment:
Why do we need to call IsReady before the Try operation in this pattern?  IsReady takes a lock to get the sequence number, but that sequence number isn't used unless we fall through to PerformSyncOperation.  PerformSyncOperation can already handle the case where the previously retrieved sequence number doesn't match, just updating its state of the world.  So couldn't we avoid the extra lock on the fast path and just defer the work to the slow path?